integration/container/nat_test.go

```diff
@@ -29,7 +29,14 @@ func TestNetworkNat(t *testing.T) {
 	startServerContainer(ctx, t, msg, 8080)
 
 	endpoint := getExternalAddress(t)
+	try := 0
+retry:
 	conn, err := net.Dial("tcp", net.JoinHostPort(endpoint.String(), "8080"))
+	if err != nil && try < 10 {
+		try++
+		time.Sleep(200 * time.Millisecond)
+		goto retry
+	}
```

Comment:
I think these changes didn't help. I ended up doing this: https://github.com/rata/moby/commit/2b723e7ff7f915244a8b1295ba5daaeaca20c747

There is a bug saying these tests are flaky (not just the one skipped, IIRC) and I just added the skip like other tests on this file do. You can verify this, but my memory is that at least not all of the changes in this file helped to make the tests green

Reply:
@kolyshkin sorry, with "these changes" I meant just the retry here in this file. The other changes are still needed, or at least I've seen them flaky with runc 1.2.0 final release in a quick test.

The only difference is that with runc 1.2.0-rcX the tests failed reliably. Now with the overlayfs it is faster and they are just flaky. But I think the rest of the changes to moby tests are needed